lib/routes/account.js

```diff
@@ -2589,35 +2590,45 @@ module.exports = (
         var form = request.payload
         var authPW = Buffer(form.authPW, 'hex')
         var uid
+        var devicesToNotify
         customs.check(
           request,
           form.email,
           'accountDestroy')
           .then(db.emailRecord.bind(db, form.email))
           .then(
             function (emailRecord) {
-              uid = emailRecord.uid.toString('hex')
+              uid = emailRecord.uid
 
               return checkPassword(emailRecord, authPW, request.app.clientAddress)
                 .then(
                   function (match) {
                     if (! match) {
                       throw error.incorrectPassword(emailRecord.email, form.email)
                     }
+                    // We fetch the devices to notify before deleteAccount() it
```

Comment:
tiny nit: there's something grammatically wrong with "before deleteAccount() it deletes all the devices", I think you either need to add a comma or remove the "it".